benchmark/src/main/java/io/servicefabric/transport/protocol/ProtostuffMessageSerializerBenchmark.java

```diff
@@ -0,0 +1,109 @@
+package io.servicefabric.transport.protocol;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.*;
+
+import com.google.common.collect.ImmutableMap;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.servicefabric.cluster.gossip.Gossip;
+import io.servicefabric.cluster.gossip.GossipRequest;
+import io.servicefabric.transport.TransportHeaders;
+
+@Fork(2)
+@State(Scope.Thread)
+@Threads(4)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class ProtostuffMessageSerializerBenchmark {
```

Comment:
Also you put headers into each Gossip so in case of message you have 1xheaders, in case of list10 - 10xheaders and list100 - 100xheaders. Since messages are generally small this headers can significantly increase releavnt sizes of messages. Actually I would say it will be helpful to output actually size of deserailized messages to see (my bet that actual gossip size will be not a 1k, but around 7k-8k).